core/sql/optimizer/RelExpr.cpp

```diff
@@ -1434,6 +1449,44 @@ void RelExpr::pushdownCoveredExpr(const ValueIdSet & outputExpr,
       computeValuesReqdForPredicates(predicatesOnParent,
                                      exprToEvalOnParent);
 
```

Comment:
I wonder if we subtract those push-down non-VEG equality predicates from the predicates to be evaluated on the parent.

Reply:
We don't today, and I have not changed this. As far as I can tell, this does not cause any issues.